src/vellum/workflows/inputs/base.py

```diff
@@ -40,14 +42,38 @@ class BaseInputs(metaclass=_BaseInputsMeta):
     __parent_class__: Type = type(None)
 
     def __init__(self, **kwargs: Any) -> None:
+        for name, field_type in self.__class__.__annotations__.items():
+            if name not in kwargs and name not in vars(self.__class__):
+                origin = get_origin(field_type)
+                args = get_args(field_type)
+                if not (origin is Union and type(None) in args):
+                    raise WorkflowInitializationException(
+                        message="Required input variables should have defined value",
+                        code=WorkflowErrorCode.INVALID_INPUTS,
+                    )
+
         for name, value in kwargs.items():
```

Comment:
[nit] we could do the `setattr` calls above and not need this loop. Doing it this way also avoids an unexpected kwarg from being set to the class